### PR TITLE
Refactors ui-select

### PR DESCRIPTION
**Pull request by pzuraq**
@Ticketfly/frontenders @djvoa12

### CODE REVIEW

File: addon/components/ui-select.js [hunk 1]
```diff
@@ -1,21 +1,54 @@
...
+      const selection = this.get('selection');
+      const optionValuePath = this.get('optionValuePath');
 
+      if (selection && optionValuePath) {
+        return get(selection, optionValuePath);
+      } else if (selection) {
+        return selection;
+      }
+    },
+    set(key, value) {
```
Comment (twokul) [inline addon/components/ui-select.js]: did you forget an additional space above the line?

File: addon/components/ui-select.js [hunk 2]
```diff
@@ -1,21 +1,54 @@
 import Ember from 'ember';
+import ComponentIdMixin from '../mixins/component-id';
 import layout from '../templates/components/ui-select';
 
-const computed = Ember.computed;
+const { computed, get } = Ember;
 
-const generateGuid = Ember.generateGuid;
+const { reads } = computed;
```
Comment (twokul) [inline addon/components/ui-select.js]: am I missing something or it is not used anywhere?

File: app/styles/ticketfly-ui/_variables.scss [hunk 3]
```diff
@@ -42,7 +42,7 @@ $gray-dark: #3D3C4A;
 $default: $white;
 $default-active: $gray-lighter;
 $default-color: $gray-med;
-$default-border: $gray-lightest;
+$default-border: #ddd;
```
Comment (twokul) [inline app/styles/ticketfly-ui/_variables.scss]: should this stay as a variable?

File: addon/components/ui-select.js [hunk 4]
```diff
@@ -1,21 +1,54 @@
...
-    this.set('value', value);
-  }
+  value: computed('selection', 'optionValuePath', {
+    get() {
+      const selection = this.get('selection');
+      const optionValuePath = this.get('optionValuePath');
 
+      if (selection && optionValuePath) {
+        return get(selection, optionValuePath);
+      } else if (selection) {
```
Comment (twokul) [inline addon/components/ui-select.js]: I think this `else if` can be replaced with just `return selection;`

File: app/styles/ticketfly-ui/_variables.scss (hunk 3 above)
Comment (pzuraq) [inline app/styles/ticketfly-ui/_variables.scss]: Going to be sitting down with the designers and finalizing colors next week. We already have 5 or so gray vars so I don't want to add another.

**Pull request merged**